### PR TITLE
feat: add way to install from remote branch or commit as well as renames [APE-1027]

### PR DESCRIPTION
**Pull request by antazoey**
fixes: #18 

Fixes: APE-1027

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -53,12 +56,16 @@ runs:
     with:
       path: |
         ${{ steps.pip-cache-dir-path.outputs.dir }}
-      key: ${{ inputs.python-version }}-pip-${{ inputs.ape-version-pin }}
+      key: ${{ inputs.python-version }}-pip-${{ inputs.ape-version }}
 
-  - run: pip install eth-ape${{ inputs.ape-version-pin }}
+  - run: |
+      if [ -n "${{ inputs.ape-remote-git-version }}" ];
```
[the remark this answers is not in the document]
Comment (antazoey) [reply action.yml:62]: ok apparently you can test the action in the same repo
surprised we dont have that yet so may just do that

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+name: tests
```
Comment (antazoey) [inline .github/workflows/test.yaml:1]: it seems im unable to test that this workflow works without merging it to main first.

**Comment by antazoey** [comment]
- [ ] add docs
- [ ] add way to install plugins from branches as well :( or at least try to, else idk how ima do this

**Comment by antazoey** [comment]
i did find a way to test this; you can use actions from branches in other repos

**Comment by antazoey** [comment]
can use with this branch to also instal remote plugins: feat/allow-installing-plugins-from-branches

```yaml
    - uses: antazoey/github-action@feat/much-upgrade
      with:
        python-version: "3.10"
        ape-remote-git-version: "https://github.com/antazoey/ape.git@both"
        ape-plugins-list: vyper@git+https://github.com/antazoey/ape-vyper.git@test/cov
```

File: .github/workflows/test.yaml (hunk 2 above)
Comment (fubuloubu) [reply .github/workflows/test.yaml:1]: Yeah we never quite figured out how to get it to work

File: action.yml [hunk 3]
```diff
@@ -7,12 +7,15 @@ branding:
 
 inputs:
   python-version: 
-    description: 'Override version of python used to run ape- default 3.8'
+    description: 'Override version of python used to run ape- default 3.10'
     required: False
-    default: '3.8'
+    default: '3.10'
```
Comment (fubuloubu) [inline action.yml:12]: I think this is the only thing that makes it a breaking change

**Review by fubuloubu — APPROVED** [review]
If we remove the 3.10 change, we can merge this and release as `v2` latest, and then do the change for 3.10 as a `v3` tag on top of that

File: action.yml (hunk 3 above)
Comment (antazoey) [reply action.yml:12]: ah ok

File: action.yml [hunk 4]
```diff
@@ -7,12 +7,15 @@ branding:
...
   ape-version-pin:
     description: 'Override version of eth-ape'
     required: False
+  ape-remote-git-version:
```
Comment (antazoey) [inline action.yml:16]: im wondering if instead we want to allow `git+` values in `ape-version-pin`, so it works the same as https://github.com/ApeWorX/ape/pull/1468

Comment (fubuloubu) [reply action.yml:16]: I don't see why not, you could probably just a switch on where it starts with `==`/`>=`/etc.

**Comment by antazoey** [comment]
I am trying to use this. I see it is installing everything correctly. The plugins actually are definitely correct. However, `eth-ape` still seems to be using an older version!! IDKW. It installs the new version but doesn't actually use it, what..in..the..heck

File: .github/workflows/test.yaml (hunk 2 above)
Comment (antazoey) [reply .github/workflows/test.yaml:1]: for test am testing by using `- uses: antazoey/github-action@feat/much-upgrad` in a different project

**Comment by antazoey** [comment]
btw my testing is happening here:  https://github.com/antazoey/ape-playground/pull/1

**Comment by antazoey** [comment]
i think it not putting it in the hosted tool cache, and the binaries in the end comes from there so even if it installs it, it wont work unless it ends up in that cache. im not sure why it's not tho, maybe the cache key doesnt like some symbols from a URL..

**Comment by antazoey** [comment]
OH DANG

I figured it out....

so `eth-ape` when installing from a remote like this comes back with version 0.1.0 as the version..
then when you install any plugin, it overrides it and sets it to the latest..

dang...
maybe can be fixed by including tags tho before installing the git+

File: action.yml [hunk 5]
```diff
@@ -11,7 +11,7 @@ inputs:
     required: False
     default: '3.8'
   ape-version-pin:
-    description: 'Override version of eth-ape'
+    description: 'Override version of eth-ape; can also use a git+ prefixed value'
```
Comment (fubuloubu) [inline action.yml:14]:
```suggestion
    description: 'Override version of eth-ape, can also use a `git+` prefixed value'
```

**Review by fubuloubu — APPROVED** [review]

**Review by fubuloubu — APPROVED** [review]

**Pull request merged**